### PR TITLE
Restructure vcpkg jobs

### PR DESCRIPTION
**Pull request by Znote**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Restructure vcpkg github action
Instead of having a consolidated job for both unix and windows, keep them as separate jobs. 

Changed from one consolidated job into two separate jobs.
One job `unix` for ubuntu and macos
One job `windows` for windows

Removed luajit=off from windows job

**Issues addressed:** 
Omitted the two failing windows jobs. (related to luajit=off). 
Two windows compile actions are already working, for debug and release, I also tested the compiling tutorial in the wiki with a success. I don't think the two failing actions are of any value to us, and is just annoying. Especially since nobody seems to care about fixing them. (I have no idea how to fix them). 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

<https://github.com/Znote/forgottenserver/actions/runs/1073447752>
image

### CODE REVIEW

**Comment by marmichalski** [comment]
You could just [exclude](https://github.com/edubart/otclient/blob/master/.github/workflows/build-vcpkg.yml#L59-L61) luajit from windows matrix. 🤔

**Comment by Znote** [comment]
> You could just [exclude](https://github.com/edubart/otclient/blob/master/.github/workflows/build-vcpkg.yml#L59-L61) luajit from windows matrix. 🤔

Yes, that would work. But I argue that this looks better in the actions overview. And now it is easier to add steps without bleeding code. (etc steps in the windows matrix will not affect unix matrix if you screw up conditional statements).

File: .github/workflows/build-vcpkg.yml [hunk 1]
```diff
@@ -145,4 +204,4 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           name: tfs-${{ matrix.name }}-${{ matrix.buildtype }}-luajit=${{ matrix.luajit }}-${{ github.sha }}
-          path: ${{ github.workspace }}
```
Comment (ranisalt) [inline .github/workflows/build-vcpkg.yml:148]: Please restore the end of file newline :pray:

File: .github/workflows/build-vcpkg.yml [hunk 2]
```diff
@@ -123,6 +111,77 @@ jobs:
...
+      - name: Unshallow
+        run: git fetch --prune --unshallow
+
+      - name: Get latest CMake
+        # Using 'latest' branch, the latest CMake is installed.
+        uses: lukka/get-cmake@latest
+
+      - name: Windows - remove C:/mysql*
+        run: rm -r -fo C:/mysql-5.7.21-winx64
+        if: contains( matrix.os, 'windows')
```
Comment (ranisalt) [inline .github/workflows/build-vcpkg.yml:158]: Won't this always be true? This matrix only runs windows

File: .github/workflows/build-vcpkg.yml [hunk 3]
```diff
@@ -123,6 +111,77 @@ jobs:
...
+  windows:
+    name: ${{ matrix.os }}-${{ matrix.cxx }}-${{ matrix.buildtype }}-luajit=${{ matrix.luajit }}
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 8
+      matrix:
+        name: [windows-msvc]
+        buildtype: [Debug, Release]
+        luajit: [on]
```
Comment (ranisalt) [inline .github/workflows/build-vcpkg.yml:133]: You can drop this from the matrix and from the job name, unless it is used for something else

File: .github/workflows/build-vcpkg.yml (hunk 1 above)
Comment (MillhioreBT) [reply .github/workflows/build-vcpkg.yml:148]: I'm not the only one who loves to leave a line at the end uwu

File: .github/workflows/build-vcpkg.yml [hunk 4]
```diff
@@ -121,21 +109,87 @@ jobs:
         with:
           name: tfs-${{ matrix.name }}-${{ matrix.buildtype }}-luajit=${{ matrix.luajit }}-${{ github.sha }}
           path: ${{ runner.workspace }}/build/tfs
-        if: "! contains( matrix.os, 'windows')"
 
-      - name: Upload artifact binary (exe)
+      - name: Prepare datapack contents
+        run: find . -maxdepth 1 ! -name data ! -name config.lua.dist ! -name key.pem ! -name LICENSE ! -name README.md ! -name schema.sql -exec rm -r {} \;
+        shell: bash
```
Comment (ranisalt) [inline .github/workflows/build-vcpkg.yml:115]: What does this new step do?

Comment (Znote) [reply .github/workflows/build-vcpkg.yml:115]: The diff checker is trolling you, if you view the raw file, the step above uploads the binary to the artifact, 
this step cleans up the runner workspace from unnecessary files such as build files, and retails data folder and config etc, 
and the step below uploads these remaining files to the artifact

See artifacts from my trigger test:
<https://github.com/Znote/forgottenserver/actions/runs/1076419470#artifacts>

Comment (ranisalt) [reply .github/workflows/build-vcpkg.yml:115]: Yeah the differ strikes again

**Review by ranisalt — APPROVED** [review]
LGTM (I don't use Windows)

File: .github/workflows/build-vcpkg.yml [hunk 5]
```diff
@@ -121,21 +109,87 @@ jobs:
...
+          echo "CXX=${{ matrix.cxx }}" >> $GITHUB_ENV
+          echo "CC=${{ matrix.cc }}" >> $GITHUB_ENV
+
+      - name: Run vcpkg
+        uses: lukka/run-vcpkg@main
+        with:
+          vcpkgArguments: ${{ matrix.packages }}
+          vcpkgDirectory: ${{ runner.workspace }}/vcpkg/
+          vcpkgTriplet: ${{ matrix.triplet }}
+          vcpkgGitCommitId: 5568f110b509a9fd90711978a7cb76bae75bb092
```
Comment (DSpeichert) [inline .github/workflows/build-vcpkg.yml:168]: We will need to bump that periodically

**Review by DSpeichert — APPROVED** [review]

**Pull request merged**